src/upload/package.rs

```diff
@@ -54,9 +54,13 @@ impl Package {
         }
 
         let package_json: String = fs::read_to_string(manifest_path.clone())?.parse()?;
-        let package: Package = serde_json::from_str(&package_json)
-            .unwrap_or_else(|_| panic!("could not parse {}", manifest_path.display()));
 
-        Ok(package)
+        serde_json::from_str::<Package>(&package_json).map_err(|e| {
+            anyhow::anyhow!(
+                "could not parse {}, may have invalid or missing `main` or `module` keys: {}",
+                manifest_path.display(),
```

Comment:
Could you add an explanation, just a quick thing like "both keys are necessary for module format"? you can copy the messages from https://github.com/cloudflare/wrangler/blob/master/src/upload/package.rs lines 18 and 32